### DISCUSSION TITLE
Http directory listing runtime errors.

### DISCUSSION THREAD

**Discussion by pseudotronics** [cesanta/mongoose#2279] [Ask questions, get answers] [answered]
I am using packed_fs.c from the [embedded filesystem](https://github.com/cesanta/mongoose/blob/master/examples/embedded-filesystem/) example along with the following relevant bit of code.

```
static void fn(struct mg_connection *c, int ev, void *ev_data, void *fn_data)
{
	if (ev == MG_EV_HTTP_MSG)
	{
		struct mg_http_message *hm = (struct mg_http_message *)ev_data;
		if (mg_http_match_uri(hm, "/api/hello"))
		{ // On /api/hello requests,
			mg_http_reply(c, 200, "", "{%m:%d}\n",
						  MG_ESC("status"), 1); // Send dynamic JSON response
		}
		else
		{														// For all other URIs,
			struct mg_http_serve_opts opts = {
				.root_dir = "/",
				.fs = &mg_fs_packed
			}; // Serve files
			mg_http_serve_dir(c, hm, &opts);
			(void)fn_data;
		}
	}
}
```

mongoose_custom.h
```
#pragma once

#define MG_ARCH MG_ARCH_FREERTOS
#define MG_ENABLE_FREERTOS_TCP 1
#define MG_IO_SIZE 256
#define MG_ENABLE_CUSTOM_RANDOM 1
#define MG_ENABLE_PACKED_FS 1
```

I am not sure what I am doing wrong, but the directory listing doesn't seem to be working as it should for me.

I get this error when navigating to the root directory ('http://0.0.0.0:8000') via the browser.

`1115   1 mongoose.c:1795:printdirentry  2 stat(//eb_root): 88`

I think the issue with the name being mangled originates from mongoose.c:890:packed_list. As for why, I am not sure. I will try stepping through to see what the core of the issue is.

Other than this, the web content serves fine when navigated to directly.

**Comment by scaprile** [comment, marked as answer]
Please put your web files in a directory and refer to that directory as the example does

**Reply by pseudotronics** [reply]
Oh so you can't serve the root of the filesystem? 
(ie "/") It has to be directory beneath root which becomes / when it is served.

I suppose in hindsight this makes a bit of sense

**Reply by scaprile** [reply]
It is not just a matter of serving but a matter of packing, as there is an extra '/' as you've seen in the logs.
